src/pybind/mgr/prometheus/module.py

```diff
@@ -1283,6 +1283,9 @@ def get_rbd_stats(self) -> None:
         # '*' can be used to indicate all pools or namespaces
         pools_string = cast(str, self.get_localized_module_option('rbd_stats_pools'))
         pool_keys = []
+        osd_map = self.get('osd_map')
+        existing_rbd_pools = [pool for pool in osd_map['pools']
+                              if 'rbd' in pool.get('application_metadata', {})]
```

Comment:
Do you need the actual pool objects?  If you generate a list of pool names instead of pool objects here, you could replace the loop in `pool_name != "*"` branch with a simple `if`.

Reply:
Also, "existing" in the variable names seems redundant -- I'd go with just `rbd_pools`.

Reply:
> Do you need the actual pool objects? If you generate a list of pool names instead of pool objects here, you could replace the loop in `pool_name != "*"` branch with a simple `if`.

IT's not just `pool_name`, `images` is also being used https://github.com/ceph/ceph/blob/main/src/pybind/mgr/prometheus/module.py#L1405

Reply:
> IT's not just `pool_name`, `images` is also being used https://github.com/ceph/ceph/blob/main/src/pybind/mgr/prometheus/module.py#L1405

Isn't that a completely different object?

Reply:
Yeah, probably! I did check something before, will have a look again and will filter only names if so, thanks!